### PR TITLE
Clean up KnativeServing lifecycle and relevant tests.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

* Removes a few unused functions.
* Reorders file a little.
* Adds documentation to all functions.
* Rewrites tests with common helpers to increase coverage.

**Release Note**

```release-note
NONE
```

/assign @jcrossley3 @houshengbo

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/test pull-knative-sandbox-operator-integration-tests

No idea how that's supposed to be related (istio yaml not found)

File: pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go [hunk 1]
```diff
@@ -31,41 +31,22 @@ func (ks *KnativeServing) GroupVersionKind() schema.GroupVersionKind {
...
-}
-
-func (is *KnativeServingStatus) IsDeploying() bool {
-	return is.IsInstalled() && !is.IsAvailable()
-}
-
-func (is *KnativeServingStatus) IsFullySupported() bool {
-	return is.GetCondition(DependenciesInstalled).IsTrue()
-}
-
```
Comment (jcrossley3) [inline pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go:53]: I liked these. :(

Comment (markusthoemmes) [reply pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go:53]: They are not used anywhere though and probably shouldn't be either :). I think we used to use these to gate reconciles which we shouldn't do.

File: pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go [hunk 2]
```diff
@@ -74,28 +55,45 @@ func (is *KnativeServingStatus) MarkInstallSucceeded() {
...
 		"NotReady",
 		"Waiting on deployments")
 }
 
+// MarkDependenciesInstalled marks the DependenciesInstalled status as true.
 func (is *KnativeServingStatus) MarkDependenciesInstalled() {
 	conditions.Manage(is).MarkTrue(DependenciesInstalled)
 }
 
+// MarkDependencyInstalling marks the MarkDependencyInstalling status as false with the
```
Comment (jcrossley3) [inline pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go:86]: typo: "...marks the DependenciesInstalled..."

File: pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go [hunk 3]
```diff
@@ -74,28 +55,45 @@ func (is *KnativeServingStatus) MarkInstallSucceeded() {
...
+// given message.
 func (is *KnativeServingStatus) MarkDependencyInstalling(msg string) {
 	conditions.Manage(is).MarkFalse(
 		DependenciesInstalled,
 		"Installing",
 		"Dependency installing: %s", msg)
 }
 
+// MarkDependencyMissing marks the MarkDependencyInstalling status as false with the
```
Comment (jcrossley3) [inline pkg/apis/operator/v1alpha1/knativeserving_lifecycle.go:95]: same typo as above

**Pull request merged**